app/js/renderer.js

```diff
@@ -21,6 +21,7 @@ asciidoctor.browser.renderer = (webExtension, document, Constants, Settings, Dom
       const settings = await Settings.getRenderingSettings();
       const asciidoctorDocument = module.convert(source, settings);
 
+      Dom.setViewport();
```

Comment:
The `update` function is called every time the content is updated so `Dom.setViewport` will be executed multiple times.
Since this function should be executed only once, we need to move it to the `loadContent` function in `app/js/loader.js` (just before calling `Renderer.update`):

https://github.com/asciidoctor/asciidoctor-browser-extension/blob/e302bae81682dbab93c730c9ec0a6c81ccf32a53/app/js/loader.js#L24-L26

While we're at it, I think we should do a little refactoring and create a function named `prepare` (or `init`) in `Renderer`:


```js
module.prepare = () => {
  module.appendMathJax();
  module.appendHighlightJsScript();
  Dom.setViewport();
};
```

```js
Renderer.prepare(); 
Renderer.update(request.responseText); 
```

What do you think ?

Reply:
Calling `Dom.setViewport` in the `update` step would be appropriate if the `content` viewport attribute was obtained from the Asciidoctor output. For this I think `header_footer` would have to be set to `true`. This would be ideal but I am not familiar enough with the API to achieve this so I will assume a fairly standard value for the `content` attribute and set that. While this constant behaviour exists, it can go in `prepare` step.

Reply:
> Calling Dom.setViewport in the update step would be appropriate if the content viewport attribute was obtained from the Asciidoctor output. 

Yes indeed 😄 

> For this I think header_footer would have to be set to true. 

Actually I've never tried to replace the entire HTML document with the Asciidoctor output (using `header_footer`). I don't know if it's a good idea or not...